### ISSUE TITLE
microsoft/chakracore

### ISSUE THREAD

**Issue by berwyn** [LoopPerfect/buckaroo-wishlist#77]
Would love to have [microsoft/chakracore](https://github.com/microsoft/chakracore) on all three major platforms!

**Comment by njlr** [comment]
Agreed! We already have a JS interpreter in [mujs](https://github.com/ccxvii/mujs), but would be good to provide The Big Three (V8, Chakra and SpiderMonkey). 

I will take a look at Chakra now... which platform(s) are you working on?

**Comment by njlr** [comment]
I was unable to build Chakra on macOS using the supplied build system (https://github.com/Microsoft/ChakraCore/issues/3905). Will pick this up again once MS respond.

**Comment by njlr** [comment]
Pending response on https://github.com/Microsoft/ChakraCore/issues/3916

**Comment by berwyn** [comment]
Yeah, I'm specifically hoping to get ChakraCore because I need a non-JIT platform. I'm targeting all three major OSes, but I haven't yet tried macOS

**Comment by njlr** [comment]
Hmmm ChakraCore is jitted, I believe. If you are looking for something very lightweight, I would suggest taking a look at https://github.com/ccxvii/mujs

**Comment by berwyn** [comment]
Chakra has both a JIT and no-JIT mode, which is determined by the presence or absence of the `--no-jit` flag at build-time

**Comment by berwyn** [comment]
Just confirmed on macOS that `./build.sh --static --no-jit --icu=/usr/local/opt/icu4c/include/ --test-build -j=4` (which is what I need for my use case) fails for 1.7.2 but works on `master` and `release/1.7`, so it looks like Microsoft/ChakraCore#3905 was resolved correctly

**Comment by njlr** [comment]
Yes, I was able to build using their script on macOS. The problem now is the include paths (https://github.com/Microsoft/ChakraCore/issues/3916)
